contrib/debian/rules

```diff
@@ -6,18 +6,44 @@
 #	$(if $(filter nocheck,$(DEB_BUILD_OPTIONS)),,src/test_bitcoin)
 
 %:
-	dh --with bash-completion $@
+	dh --with bash-completion --with systemd $@
 
 override_dh_auto_clean:
 	if [ -f Makefile ]; then $(MAKE) distclean; fi
 	rm -rf Makefile.in aclocal.m4 configure src/Makefile.in src/bitcoin-config.h.in src/build-aux src/qt/Makefile.in src/qt/test/Makefile.in src/test/Makefile.in
 
 QT=$(shell dpkg-vendor --derives-from Ubuntu && echo qt4 || echo qt5)
+# qt4 is very broken on arm
+ifeq ($(findstring arm,$(shell uname -m)),arm)
+	QT=qt5
+endif
+ifeq ($(findstring aarch64,$(shell uname -m)),aarch64)
+	QT=qt5
+endif
 
 # Yea, autogen should be run on the source archive, but I like doing git archive
 override_dh_auto_configure:
 	./autogen.sh
 	./configure --with-gui=$(QT)
 
 override_dh_auto_test:
+ifeq ($(QT), qt4)
+	xvfb-run -n 99 -l make check
+else
 	make check
+endif
+
+# No SysV or Upstart init scripts included
+override_dh_installinit:
+	dh_installinit \
+		--noscripts
+
+# Don’t enable service by default
+override_dh_systemd_enable:
+	dh_systemd_enable \
+		--no-enable
+
+# Restart after upgrade
+override_dh_systemd_start:
+	dh_systemd_start \
+		--restart-after-upgrade
```

Comment:
Why? I dont think anyone would be surprised by their bitcoind going down prior to upgrade instead of only being restarted, and this could break eg a lxc VM as you'll get a text file busy error.

Reply:
@TheBlueMatt, good point.

`--restart-after-upgrade` is default in `dh_systemd_start` from compat level 10, so the reason is perhaps best described as emulating modern defaults. I agree that for bitcoind, there should not be any real issues with using the defaults.

I don't quite follow the problems you mention though. As far as I can tell, an upgrade should work perfectly, even when leaving the daemon running during the upgrade. Nothing runtime related is touched during the package upgrade, only the new `bitcoind` and `bitcoin-cli` binaries are put into place, making them ready for the late restart. Unless I'm missing something, this should be no different when running in a container?

If you feel there's a risk here, I'll happily remove this!

Reply:
Eh, if its default whatever, but linux generally refuses to replace binaries on disk if they are running inside of an LXC/docker container (though will happily do so during normal operation).